src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/Manage Statistics/ManageStatisticsCardView.swift

```diff
@@ -0,0 +1,167 @@
+
+		self.onAddStateButtonTap = onAddStateButtonTap
+		self.onAddDistrict = onAddDistrict
+		self.onDismissState = onDismissState
+		self.onDismissDistrict = onDismissDistrict
+		self.onFetchGroupData = onFetchGroupData
+		self.onEditButtonTap = onEditButtonTap
+		self.onAccessibilityFocus = onAccessibilityFocus
+
+		updateUI(for: availableCardsState)
+	}
+
+	@IBAction func onAddLocalIncidenceButtonPressed(_ sender: Any) {
+		guard let model = self.model else {
+			Log.warning("AddStatistics model is nil", log: .localStatistics)
+			return
+		}
+		viewModel = ManageStatisticsCardsViewModel(
+			localStatisticsModel: model,
+			presentFederalStatesList: { selectedStateValueViewModel in
+				self.presentAddLocalStatistics(selectValueViewModel: selectedStateValueViewModel)
+			},
+			presentSelectDistrictsList: { selectedDistrictValueViewModel in
+				self.presentAddLocalStatisticsDistrict(selectValueViewModel: selectedDistrictValueViewModel)
+				
+			}, onFetchGroupData: { [weak self] district in
+				self?.onFetchGroupData?(district)
+			}
+		)
+		viewModel?.presentStateSelection()
+	}
+
+	func updateUI(for state: LocalStatisticsState) {
+		// clear
+		stackView.arrangedSubviews.forEach { subview in
+			stackView.removeArrangedSubview(subview)
+			subview.removeFromSuperview()
+		}
+
+		let addView = { () -> CustomDashedView in
+			let add = CustomDashedView.instance(for: .add)
+			add.tapHandler = { [weak self] in
+				Log.debug("add…", log: .ui)
+				self?.onAddLocalIncidenceButtonPressed(add)
+			}
+			add.label.onAccessibilityFocus = onAccessibilityFocus
+			return add
+		}()
+
+		let modifyView = { () -> CustomDashedView in
+			let modify = CustomDashedView.instance(for: .modify)
+			modify.tapHandler = { [weak self] in
+				Log.debug("modify…", log: .ui)
+				self?.onEditButtonTap?()
+			}
+			modify.label.onAccessibilityFocus = onAccessibilityFocus
+			return modify
+		}()
+
+		switch state {
+		case .empty:
+			// just 'add'
+			stackView.addArrangedSubview(addView)
+		case .notYetFull:
+			// 'add' & 'modify'
+			stackView.addArrangedSubview(addView)
+			stackView.addArrangedSubview(modifyView)
+		case .full:
+			// just 'modify'
+			stackView.addArrangedSubview(modifyView)
+		}
+	}
+	
+	// MARK: - Private
+
+	private func presentAddLocalStatistics(selectValueViewModel: SelectValueViewModel) {
+		let selectValueViewController = SelectValueTableViewController(
+			selectValueViewModel,
+			closeOnSelection: false,
+			dismiss: { [weak self] in
+				self?.onDismissState?()
+			})
+		onAddStateButtonTap?(selectValueViewController)
+	}
+
+	private func presentAddLocalStatisticsDistrict(selectValueViewModel: SelectValueViewModel) {
+		let selectValueViewController = SelectValueTableViewController(
+			selectValueViewModel,
+			closeOnSelection: true,
+			dismiss: { [weak self] in
+				let dismissToRoot = self?.viewModel?.district != nil
+				self?.onDismissDistrict?(dismissToRoot)
+			}
+		)
+		onAddDistrict?(selectValueViewController)
+	}
```

Comment:
View controllers should not be instantiated inside views, that is the responsibility of the coordinator